evaluation.py

```diff
@@ -0,0 +1,113 @@
+import argparse
+from hourglass import HourglassNet
+from data_generator import DataGenerator
+from constants import *
+from util import *
+
+import time
+from datetime import timedelta
+import matplotlib.pyplot as plt
+import numpy as np
+from PIL import Image
+import glob
+import cv2
+
+def predict(hourglass, model_json, model_weights, subset):
+    hgnet = HourglassNet(num_classes=NUM_COCO_KEYPOINTS, num_stacks=hourglass, num_channels=NUM_CHANNELS, inres=INPUT_DIM,
+                            outres=OUTPUT_DIM)
+    hgnet._load_model(model_json, model_weights)
+
+    val_df = hgnet.load_and_filter_annotations(DEFAULT_TRAIN_ANNOT_PATH, DEFAULT_VAL_ANNOT_PATH, subset)
+
+    generator = DataGenerator(
+        df=val_df,
+        base_dir=DEFAULT_TRAIN_IMG_PATH,
+        input_dim=INPUT_DIM,
+        output_dim=OUTPUT_DIM,
+        num_hg_blocks=DEFAULT_NUM_HG,
+        shuffle=False,  
+        batch_size=1,
+        online_fetch=False)
+
+    # Select image to predict heatmaps
+    X_batch, y_stacked = generator[168]
+    X = X_batch[0] # take first example of batch
+
+    X = X.reshape(1, 256, 256, 3) # predict needs shape (1, 256, 256, 3)
```

Comment:
We should figure out why this reshaping is necessary. I would have expected the model to be able to take a single image of shape `(256, 256, 3)`. Is the 1 related the batch size that was used when training the model? If so that seems werid.

Reply:
yeah I think we need the reshape due to batch size. Although I think `numpy.expand_dims` is better than a reshape here.

Reply:
So I tried testing this with batch size not equal to 1 and it still needs the shape to be (1, 256, 256, 3) for some reason

Reply:
@nicolepeverley have you tried on different models that have been trained with different batch sizes? In all cases is it always 1?

Reply:
It's because the keras model can process multiple images at once [Source](https://stackoverflow.com/questions/50723287/meaning-of-batch-size-in-model-evaluate):
> The evaluate function of Model has a batch size just in order to speed-up evaluation, as the network can process multiple samples at a time, and with a GPU this makes evaluation much faster. I think the only way to reduce the effect of this would be to set batch_size to one.

Reply:
Oh perfect that explains it then ty

Reply:
(I didn't know this until now haha)